testing/spectest/shared/common/forkchoice/runner.go

```diff
@@ -293,10 +296,160 @@ func runBlobStep(t *testing.T,
+					Index:             uint64(columnIndex),
+					Column:            [][]byte{},
+					KzgCommitments:    kzgs,
+					KzgProofs:         make([][]byte, 0),
+					SignedBlockHeader: sh,
+				}
+			}
+		} else {
+			numCells := len(kzgs)
+			column := make([][]byte, numCells)
+			for cellIndex := 0; cellIndex < numCells; cellIndex++ {
+				cell := make([]byte, 2048)
+				cellStart := cellIndex * 2048
+				cellEnd := cellStart + 2048
+				if cellEnd <= len(dataColumnSSZ) {
+					copy(cell, dataColumnSSZ[cellStart:cellEnd])
+				}
+				column[cellIndex] = cell
+			}
+
+			inclusionProof, err := blocks.MerkleProofKZGCommitments(block.Body())
+			require.NoError(t, err)
+
+			pb = &ethpb.DataColumnSidecar{
+				Index:                        uint64(columnIndex),
+				Column:                       column,
+				KzgCommitments:               kzgs,
+				SignedBlockHeader:            sh,
+				KzgCommitmentsInclusionProof: inclusionProof,
+			}
+		}
+
+		ro, err := blocks.NewRODataColumnWithRoot(pb, root)
+		require.NoError(t, err)
+		allColumns = append(allColumns, ro)
+	}
+
+	if len(allColumns) > 0 {
+		ini, err := builder.vwait.WaitForInitializer(context.Background())
+		require.NoError(t, err)
+		// Use different verification requirements based on whether this is a valid or invalid test case
+		var forkchoiceReqs []verification.Requirement
+		if step.Valid != nil && !*step.Valid {
+			forkchoiceReqs = verification.SpectestDataColumnSidecarRequirements
+		} else {
+			forkchoiceReqs = []verification.Requirement{
+				verification.RequireNotFromFutureSlot,
+				verification.RequireSlotAboveFinalized,
+				verification.RequireValidProposerSignature,
+				verification.RequireSidecarParentSlotLower,
+				verification.RequireSidecarDescendsFromFinalized,
+				verification.RequireSidecarInclusionProven,
+				verification.RequireSidecarProposerExpected,
+			}
+		}
+		dv := ini.NewDataColumnsVerifier(allColumns, forkchoiceReqs)
+		ctx := context.Background()
+
+		if step.Valid != nil && !*step.Valid {
+			if err := dv.ValidFields(); err != nil {
+				t.Logf("ValidFields error: %s", err.Error())
+			}
+		}
+
+		if err := dv.NotFromFutureSlot(); err != nil {
+			t.Logf("NotFromFutureSlot error: %s", err.Error())
+		}
+		if err := dv.SlotAboveFinalized(); err != nil {
+			t.Logf("SlotAboveFinalized error: %s", err.Error())
+		}
+		if err := dv.SidecarInclusionProven(); err != nil {
+			t.Logf("SidecarInclusionProven error: %s", err.Error())
+		}
+		if err := dv.ValidProposerSignature(ctx); err != nil {
+			t.Logf("ValidProposerSignature error: %s", err.Error())
+		}
+		if err := dv.SidecarParentSlotLower(); err != nil {
+			t.Logf("SidecarParentSlotLower error: %s", err.Error())
+		}
+		if err := dv.SidecarDescendsFromFinalized(); err != nil {
+			t.Logf("SidecarDescendsFromFinalized error: %s", err.Error())
+		}
+		if err := dv.SidecarProposerExpected(ctx); err != nil {
+			t.Logf("SidecarProposerExpected error: %s", err.Error())
+		}
```

Comment:
Shouldn't these be failures?

Reply:
`VerifiedRODataColumns` checks it via `!dv.results.allSatisfied() `